.github/workflows/integration.yml

```diff
@@ -0,0 +1,243 @@
+        EXPECTED=12
+        until [ $(./build/fleetctl get hosts --json | wc -l | tee hostcount) -ge $EXPECTED ]; do
+          echo -n "Waiting for hosts to enroll: "
+          cat hostcount | xargs echo -n
+          echo " / $EXPECTED"
+          sleep 10
+        done
+        ./build/fleetctl get hosts
+        echo "Success! $EXPECTED hosts enrolled."
+
+    - name: Cleanup tunnel
+      if: always()
+      run: cloudflared tunnel --origincert cert.pem delete --force ${{ needs.gen.outputs.subdomain }}    
+
+  login:
+   runs-on: ubuntu-latest
+   needs: gen
+   outputs:
+     token: ${{ steps.login.outputs.token }}
+   steps:
+    # Download fleet and fleetctl binaries from last successful build on main
+    - name: Download binaries
+      uses: dawidd6/action-download-artifact@09385b76de790122f4da9c82b17bccf858b9557c #v2.16.0
+      with:
+        workflow: build-binaries.yaml
+        branch: main
+        name: build
+        path: build
+        check_artifacts: true
+    
+    # Login only here and share the token because otherwise we could hit rate limits.
+    - id: login
+      name: Attempt login
+      timeout-minutes: 5
+      run: |
+        chmod +x ./build/fleetctl
+        ./build/fleetctl config set --address ${{ needs.gen.outputs.address }}
+        until ./build/fleetctl login --email admin@example.com --password admin123#
+        do
+          echo "Retrying in 5s..."
+          sleep 5
+        done
+        TOKEN=$(cat ~/.fleet/config| grep token | awk '{ print $2 }')
+        echo "::set-output name=token::$TOKEN"
+      
+  orbit-macos:
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        orbit-channel: [ 'stable', 'edge' ]
+        osqueryd-channel: ['stable', 'edge' ]
+    runs-on: macos-latest
+    needs: [gen, login]
+    steps:
+    - name: Install dependencies
+      run: |
+        npm install -g fleetctl
+        fleetctl config set --address ${{ needs.gen.outputs.address }} --token ${{ needs.login.outputs.token }}
+
+    - name: Install Orbit
+      run: |
+        sudo hostname macos-orbit-${{ matrix.orbit-channel }}-osqueryd-${{ matrix.osqueryd-channel }}
+        SECRET_JSON=$(fleetctl get enroll_secret --json --debug)
+        echo $SECRET_JSON
+        SECRET=$(echo $SECRET_JSON | jq -r '.spec.secrets[0].secret')
+        echo "Secret: $SECRET"
+        echo "Hostname: $(hostname -s)"
+        fleetctl package --type pkg --fleet-url=${{ needs.gen.outputs.address }} --enroll-secret=$SECRET --orbit-channel=${{ matrix.orbit-channel }} --osqueryd-channel=${{ matrix.osqueryd-channel }}
+        sudo installer -pkg fleet-osquery.pkg -target /
+        until fleetctl get hosts | grep -iF $(hostname -s);
+        do
+          echo "Awaiting enrollment..."
+          sleep 1
+        done
+
+  orbit-ubuntu:
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        orbit-channel: [ 'stable', 'edge' ]
+        osqueryd-channel: ['stable', 'edge' ]
```

Comment:
So for both orbit and osquery stable is the latest official release and edge is the latest build from master/main, correct?

Reply:
I wonder if we want to test orbit edge with osquery stable too.

Reply:
This strategy will run all four combinations, see left sidebar on https://github.com/fleetdm/fleet/actions/runs/1680553777.

Definitions of stable/edge are still TBD. The way I've been using them so far is to push new versions to `edge`, do some testing, then push them to `stable`.